core/templates/services/context.service.spec.ts

```diff
@@ -64,6 +64,15 @@ describe('Context service', () => {
       expect(ecs.getQuestionPlayerIsManuallySet()).toEqual(false);
       expect(ecs.isInQuestionPlayerMode()).toEqual(false);
     });
+
+    it('should correctly check the page context', ()=> {
+      expect(ecs.isInExplorationContext()).toBe(true);
+    });
+
+    it('should correctly set and retrive the exploration id', () => {
+      ecs.setExplorationId('1000');
+      expect(ecs.getExplorationId()).toBe('1000');
+    });
```

Comment:
Please don't add new setter methods to reach certain lines of code for coverage purpose. This will not help in catching regressions.

Reply:
For testing the first `if` statement in `getExplorationId()`. Perhaps make changes such that the preconditions for explorationId to be set is met before calling `getExplorationId()`. You can look at how this method is actually used in the codebase to understand its working better.

Reply:
@kevintab95 

As ContextService is the place where getExplorationId() is defined, now whenever any instance of ContextService is created then as per the constructor, explorationId would always be null. 

Now as per what it seems to me, explorationId is set by spyOn as pathname to /abcd/'123' where 123 is the string extracted. I don't find any other way where explorationId is set in ContextService. 

Can you please help me find out any possible case?

Reply:
Removed setExplorationId method.

Now, I feel there's no need of first `if` condition as its never going to be executed. Not sure if there's anyway.

Reply:
Called getExplorationId() twice so it returns the cached value on second time passing the if condition. Thanks for pointing out. Done!